bitnami/harbor/templates/chartmuseum/chartmuseum-dpl.yaml

```diff
@@ -149,18 +113,64 @@ spec:
             - name: INTERNAL_TLS_TRUST_CA_PATH
               value: /etc/harbor/ssl/chartmuseum/ca.crt
             {{- end }}
-            {{- if .Values.chartmuseum.extraEnvVars }}
-            {{- include "common.tplvalues.render" (dict "value" .Values.chartmuseum.extraEnvVars "context" $) | nindent 12 }}
-            {{- end }}
             - name: BASIC_AUTH_PASS
               valueFrom:
                 secretKeyRef:
                   # Take the password from the core component secret
                   name: {{ include "harbor.core" . }}
                   key: secret
+            {{- if .Values.chartmuseum.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.chartmuseum.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            - configMapRef:
+                name: {{ printf "%s-envvars" (include "harbor.chartmuseum" .) }}
+            - secretRef:
+                name: {{ printf "%s-secret" (include "harbor.chartmuseum" .) }}
+            {{- if .Values.chartmuseum.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.chartmuseum.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.chartmuseum.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.chartmuseum.extraEnvVarsSecret "context" $) }}
+            {{- end }}
           ports:
-            - containerPort: {{ include "harbor.chartmuseum.containerPort" .  }}
-              name: http
+            - containerPort: {{ ternary .Values.chartmuseum.containerPorts.https .Values.chartmuseum.containerPorts.http .Values.internalTLS.enabled }}
+              name: {{ ternary "https" "http" .Values.internalTLS.enabled }}
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.chartmuseum.startupProbe.enabled }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.chartmuseum.startupProbe "enabled") "context" $) | nindent 12 }}
+            tcpSocket:
```

Comment:
The use of `tcpSocket` in the startup probe is intentional?

Reply:
Yes! I think it's enough for this kind of microservices to check there's a process listening to consider it "started". Then, you have the readiness probe to actually ensure it's ready to accept connections.